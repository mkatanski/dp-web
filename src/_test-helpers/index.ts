export * from "./withResponsiveContext";
export * from "./withThemeProvider";
