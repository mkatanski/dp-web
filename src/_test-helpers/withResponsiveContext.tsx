import React from "react";
import { Context as ResponsiveContext } from "react-responsive";
import { Breakpoints } from "config/breakpoints";

export const withMobileSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.mobile }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withTabletSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.tablet }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.desktop }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withLargeDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.largeDesktop }}>
    {Component}
  </ResponsiveContext.Provider>
);
