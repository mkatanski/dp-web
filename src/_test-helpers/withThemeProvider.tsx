import React from "react";
import { ThemeProvider as DefaultThemeProvider } from "lib/theme/ThemeProvider";
import { StyledTheme, Theme } from "styles/theme/_types";
import { createMuiTheme } from "@material-ui/core";

const testTheme: StyledTheme = {
  primary: "primary",
  secondary: "secondary",
  textPrimary: "textPrimary",
  textSecondary: "textSecondary",
  textInverse: "textInverse"
};

const muiTheme = createMuiTheme({
  direction: "ltr"
});

const theme: Theme = Object.assign(testTheme, muiTheme);

export const withThemeProvider = (Component: React.ReactNode) => (
  <DefaultThemeProvider theme={theme}>{Component}</DefaultThemeProvider>
);
