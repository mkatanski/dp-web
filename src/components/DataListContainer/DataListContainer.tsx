import React from "react";
import {
  Card as MuiCard,
  CardContent,
  CardHeader,
  Divider,
  CardActions as BareCardActions
} from "@material-ui/core";
import styled from "styled-components";

import { RecordsInfo } from "components/RecordsInfo";
import { DataListPagination } from "components/DataListPagination";

export type DataListContainerProps = React.PropsWithChildren<{
  title: string;
}>;

const Card = styled(MuiCard)`
  flex: 1 1 auto;
  display: flex;
  flex-direction: column;
`;

const StyledCardContent = styled(CardContent)`
  flex: 1 1 auto;
  display: flex;
  padding: 0;

  & > div {
    overflow: scroll;
    flex: 1 1 auto;
    position: relative;
  }
`;

const CardActions = styled(BareCardActions)`
  justify-content: flex-end;
  padding: 0;
`;

export const DataListContainer: React.FC<DataListContainerProps> = ({
  children,
  title
}: DataListContainerProps) => (
  <>
    <RecordsInfo />
    <Card>
      <CardHeader title={title} />
      <Divider />
      <StyledCardContent>{children}</StyledCardContent>
      <CardActions>
        <DataListPagination />
      </CardActions>
    </Card>
  </>
);
