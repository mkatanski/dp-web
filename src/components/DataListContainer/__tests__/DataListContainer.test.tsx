/* eslint-disable react/display-name */
import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";
import { withThemeProvider } from "_test-helpers";

import { DataListContainer } from "../DataListContainer";

jest.mock("components/DataListPagination", () => ({
  DataListPagination: ({ children, ...rest }: React.PropsWithChildren<{}>) => (
    <div id="DataListPagination" data-props={rest}>
      {children}
    </div>
  )
}));

jest.mock("components/ToggleableDrawer", () => ({
  ...require.requireActual("components/ToggleableDrawer"),
  ToggleableDrawer: ({
    children,
    className,
    ...rest
  }: React.PropsWithChildren<{ className: string }>) => (
    <div id="ToggleableDrawer" className={className} data-props={rest}>
      {children}
    </div>
  )
}));

jest.mock("components/RecordsInfo", () => ({
  ...require.requireActual("components/RecordsInfo"),
  RecordsInfo: ({
    children,
    className,
    ...rest
  }: React.PropsWithChildren<{ className: string }>) => (
    <div id="RecordsInfo" className={className} data-props={rest}>
      {children}
    </div>
  )
}));

jest.mock("components/DeploymentForm", () => ({
  ...require.requireActual("components/DeploymentForm"),
  DeploymentForm: ({
    children,
    className,
    ...rest
  }: React.PropsWithChildren<{ className: string }>) => (
    <div id="DeploymentForm" className={className} data-props={rest}>
      {children}
    </div>
  )
}));

describe("components", () => {
  describe("DataListContainer", () => {
    it("should render DataListContainer with records data ", () => {
      const tree = TestRenderer.create(
        withThemeProvider(
          <DataListContainer title="test_title">Some childs</DataListContainer>
        )
      );

      expect(tree).toMatchSnapshot();
    });
  });
});
