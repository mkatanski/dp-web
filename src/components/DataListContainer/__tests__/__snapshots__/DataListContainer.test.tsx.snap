// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components DataListContainer should render DataListContainer with records data  1`] = `
Array [
  <div
    data-props={Object {}}
    id="RecordsInfo"
  />,
  .c0 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
}

.c1 > div {
  overflow: scroll;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  position: relative;
}

.c2 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 0;
}

<div
    className="c0"
    data-props={Object {}}
    id="MUI.Card"
  >
    <div
      data-props={
        Object {
          "title": "test_title",
        }
      }
      id="MUI.CardHeader"
    />
    <div
      data-props={Object {}}
      id="MUI.Divider"
    />
    <div
      className="c1"
      data-props={Object {}}
      id="MUI.CardContent"
    >
      Some childs
    </div>
    <div
      className="c2"
      data-props={Object {}}
      id="MUI.CardActions"
    >
      <div
        data-props={Object {}}
        id="DataListPagination"
      />
    </div>
  </div>,
]
`;
